internal/server/handlers.go

```diff
@@ -34,22 +34,21 @@ type API struct {
 func (a *API) CreateToken(c *gin.Context, r *api.EmptyRequest) (*api.CreateTokenResponse, error) {
 	rCtx := getRequestContext(c)
 
-	if rCtx.Authenticated.User != nil {
-		err := a.UpdateIdentityInfoFromProvider(rCtx)
-		if err != nil {
-			// this will fail if the user was removed from the IDP, which means they no longer are a valid user
-			return nil, fmt.Errorf("%w: failed to update identity info from provider: %s", internal.ErrUnauthorized, err)
-		}
-
-		token, err := access.CreateToken(rCtx)
-		if err != nil {
-			return nil, err
-		}
+	if rCtx.Authenticated.User == nil {
+		return nil, errMissingAuthentication
+	}
```

Comment:
I flipped the condition here to make it a guard class, so the diff is not as easy to follow. The previous error message was at the end of the function.

Reply:
this should never happen, it should fail at the authentication middleware, so I'm wondering if we should return a 500 error instead

Reply:
That seems reasonable to me. There are 2 other uses of `errMissingAuthentication` that I think are in the same category.

Reply:
Ok, I've made this change. These are now 500 errors, since they should only happen on programming errors.